applications/navi/doc/README.md

```diff
@@ -0,0 +1,88 @@
+3. Use the cb\_push\_notification\_subscriptions documents to determine the device to send the notification to.
+4. Call navi\_module\_sup:push/4, which will begin the process of sending the push notification (Encapsulated in navi\_listener:do\_notifications/3)
+	* do\_notifications takes three arguments. 
+		* A list of push\_notification\_subscription documents representing the registrations for each device that should receive this notification
+		* The message to appear in the body of the push notification
+		* A proplist containing any extra parameters you want to be sent in the notification (You may need to modify nv_apns and nv_fcm so it formats your custom data in the notification payload how you want).
+
+### navi\_module\_sup.erl
+This module is the supervisor for most of the activity in navi. On startup, it will create a navi\_notification\_server\_sup for each app in the system\_config.navi document, which will in turn start the
+process that actually sends the push notifications. It is also the interface for sending notifications. To send a push notification, simply call:
+``navi_module_sup:push(RegistrationId, AppName, NotificationType, Message, ExtraParameters).``
+where:
+* RegistrationId is the device token of the device to send notifications to
+* AppName is the name of the app to deliver the notification (as defined in the system\_config.navi document)
+* NotificationType is either <<"apns">> or <<"fcm">>, denoting which kind of notification server to use
+* Message is the text you want to appear in the body of the notification
+* ExtraParameters contains any sort of metadata you want to pass to the receiving application such as badge count, subtopics or other data specific to the app
+
+### navi\_notification\_server\_sup
+This module superverises one individual notification server and passes the push notification request from navi\_module\_sup to the actual notification server.
+Many instances of the module are spawned by navi\_module\_sup with arguments defining which app the notification server is for. As long as system\_config.navi has all the apps
+that you want in it, you will not need to interface with this module.
+
+### nv\_apns
+This module contains the logic for sending push notifications to iOS apps. You will not need to interact with this module if everything is set up properly unless you want to add
+custom data to your notifications.
+
+### nv\_fcm
+This module contains the logic for sending push notifications to Android apps. You will not need to interact with this module if everything is set up properly unless you want to add
+custom data to your notifications.
+
+### kapi\_navi
+This module allows for applications other than navi to push notifications. It exposes a service which can make requests through amqp to Navi. It allows for pushing to a specific device and to pushing to all devices owned by a specific user.
```

Comment:
Ok yes, this should be moved to core and renamed. If it stays here and other applications use it, those applications will then depend on having navi installed and running on the same VM as the application. This could conflict with how folks choose to deploy Kazoo. No Kazoo application should depend on another being present in the same VM - all comms should be done via AMQP.

Reply:
I have renamed the kapi module kapi_push_notifications and moved it into kazoo_amqp/api, changed the event name to push and the event categories to device, and user. I've also updated navi_listener so that it will pick up the new events.